contributing.md

```diff
@@ -0,0 +1,163 @@
+# Contributing to pylabrad
+
+## Process
+
+The following instructions are for pylabrad contributors.
+Other users working in their own fork can will use whatever procedure they like for managing contributions.
+Once those contributions are ready to be included in the main pylabrad repository, they should file a pull request and allow the pylabrad contributors to carry out the rest of the steps.
+
+1. Most changes to the code start when a user makes an issue in the pylabrad issue tracker.
+This issue explains what feature is to be added or what bug is to be fixed.
+The issue is automatically assigned a unique number
+
+1. Development proceeds in a branch
+ * The branch name is `XX-description-of-issue` where `XX` is the issue number and everything else is a brief description of the issue.
+For example, the issue filed in which we stated the need for this contributing document was #134, so the branch was called `134-contributing`.
```

Comment:
given that you don't necessarily need an issue to start writing code (see above), this branch naming convention may not always be appropriate. Let's call out the `u/<user>/<branch-name>` convention as an alternate style.